src/vs/workbench/parts/files/browser/views/explorerViewer.ts

```diff
@@ -69,16 +94,35 @@ export class FileDataSource implements IDataSource {
 	}
 
 	public hasChildren(tree: ITree, stat: FileStat | Model): boolean {
-		return stat instanceof Model || (stat instanceof FileStat && stat.isDirectory);
+		if (stat instanceof Model) {
```

Comment:
If nesting is enabled and then disabled, I don't think that will reset `isVirtualDirectory`.